### PR TITLE
reject empty readme file & fix error message for invalid extension

### PR DESCRIPTION
**Pull request by lyndaidaii**
Summary of the changes (in less than 80 characters):

reject empty readme file: 
emptyReadme

correct error message for invalid extension: 
invalidextension

Addresses https://github.com/NuGet/Engineering/issues/3707, https://github.com/NuGet/NuGetGallery/issues/8460

### CODE REVIEW

**Comment by zhhyu** [comment]
Should we also consider fixing the JS rather than just reject this kind of packages?

**Comment by lyndaidaii** [comment]
> Should we also consider fixing the JS rather than just reject this kind of packages?

Main reason to reject empty pakcage becasue we want to be consistent with what client validation during pack.

File: src/NuGetGallery/Strings.resx [hunk 1]
```diff
@@ -1166,7 +1166,7 @@ The {1} Team</value>
     <value>The &lt;readme&gt; element is not currently supported.</value>
   </data>
   <data name="UploadPackage_InvalidReadmeFileExtension" xml:space="preserve">
-    <value>The readme file has an invalid extension '{0}'. Extension must be one of the following: {1}.</value>
+    <value>The readme file has an invalid extension '{0}'. Extension must be end in: {1}.</value>
```
Comment (loic-sharma) [inline src/NuGetGallery/Strings.resx:1169]: What do you think of this?

```suggestion
    <value>The readme file has an invalid extension '{0}'. The extension must be: {1}.</value>
```

File: src/NuGetGallery/Strings.resx [hunk 2]
```diff
@@ -1215,4 +1215,7 @@ The {1} Team</value>
   <data name="UploadPackage_OwnerlessIdNamespaceConflictHtml" xml:space="preserve">
     <value>The package ID is reserved. You can upload your package with a different package ID. Reach out to &lt;a href="mailto:support@nuget.org"&gt;support@nuget.org&lt;/a&gt; if you have questions.</value>
   </data>
+  <data name="ReadmeErrorEmpty" xml:space="preserve">
+    <value>The readme file '{0}' cannot be empty</value>
```
Comment (loic-sharma) [inline src/NuGetGallery/Strings.resx:1219]: Should this end with a period?

```suggestion
    <value>The readme file '{0}' cannot be empty.</value>

```

**Review by loic-sharma — APPROVED** [review]

File: src/NuGetGallery/Services/PackageMetadataValidationService.cs [hunk 3]
```diff
@@ -458,6 +458,16 @@ private async Task<PackageValidationResult> CheckReadmeMetadataAsync(PackageArch
             }
 
             var readmeFileEntry = nuGetPackage.GetEntry(readmeFilePath);
+
+            if (readmeFileEntry.Length == 0)
```
Comment (joelverhagen) [inline src/NuGetGallery/Services/PackageMetadataValidationService.cs:462]: What about a readme with just whitespace? Would it be worth rejecting those too?

Comment (lyndaidaii) [reply src/NuGetGallery/Services/PackageMetadataValidationService.cs:462]: good question, probably not rejecting those? @chgill-MSFT  what do you think?

Comment (zhhyu) [reply src/NuGetGallery/Services/PackageMetadataValidationService.cs:462]: We should keep a consistent behavior on validations between the server and client.
If the client accepts a whitespace readme but the server rejects it, it's not a good customer experience.

Comment (joelverhagen) [reply src/NuGetGallery/Services/PackageMetadataValidationService.cs:462]: Our validations are already a superset of client, by design. I don't think an empty string README is useful. But also, it's the author's fault if they do this. My guess it is likely unintentional and not a helpful content to the user. But I don't feel strongly.

**Pull request merged**